beat/heart.py

```diff
@@ -1285,22 +1285,25 @@ class PolarityTarget(gf.meta.Receiver):
     phase = String.T(default='any_P', optional=True, 
                      help='First arrival of which seismic wave')
     
-    def update_target(self, source, engine):
+    def update_target(self, engine, source, event=None):
```

Comment:
whats your motivation behind adding event here, I think its not needed? the source will always have all the event information contained!

Reply:
> whats your motivation behind adding event here, I think its not needed? the source will always have all the event information contained!

Well, that was exactly a bug, when we tried to update targets with any sources, we just change the observed azimuths and take-off angles in the init composite that we calculated polarities with. So, we never have observed data actually. Now, synthetics data are confirming it. everything is working correctly, Today, I will run real data tests.